src/constants/checkout.ts

```diff
@@ -70,23 +70,33 @@ export const PlanDetailsLoginPageSchema = (constraints: CheckoutContextFieldCons
     .max(255, 'Maximum 255 characters'),
 }));
 
-export const PlanDetailsRegisterPageSchema = () => (z.object({
+export const PlanDetailsRegisterPageSchema = (constraints) => (z.object({
   adminEmail: z.string().trim()
     .email()
-    .min(1, 'Email is required')
-    .max(254),
+    .min(
+      constraints?.adminEmail?.minLength,
+      constraints?.adminEmail?.minLength
+        ? 'Email is required'
+        : undefined,
+    )
+    .max(constraints?.adminEmail?.maxLength),
   fullName: z.string().trim()
-    .min(1, 'Full name is required')
-    .max(255),
+    .min(
+      constraints?.fullName?.minLength,
+      constraints?.fullName?.minLength
+        ? 'Full name is required'
+        : undefined,
+    )
+    .max(constraints?.fullName?.maxLength),
   username: z.string().trim()
     .min(2, 'Username must be between 2 and 30 characters long.')
     .max(30, 'Username must be between 2 and 30 characters long.'),
   password: z.string()
-    .min(2, 'This password is too short. It must contain at least 2 characters.')
-    .max(75, 'This password is too long. It must contain no more than 75 characters.'),
+    .min(8, 'This password is too short. It must contain at least 8 characters.')
+    .max(100, 'This password is too long. It must contain no more than 100 characters.'),
   confirmPassword: z.string()
     .min(8, 'Please confirm your password')
-    .max(75, 'This password is too long. It must contain no more than 75 characters.'),
+    .max(100, 'This password is too long. It must contain no more than 100 characters.'),
```

Comment:
For some reason these changes are overlapping with https://github.com/edx/frontend-app-enterprise-checkout/pull/102 from Kira.  They both do good things, but aren't 100% on their own.  Maybe combine the solutions into this PR and close the other one.
```suggestion
password: z.string()
    .min(8, 'Password must be at least 8 characters.')
    .max(100, 'Password must be no more than 100 characters.'),
    .refine((value) => /[0-9]/.test(value), 'Password must contain at least one digit.'),
confirmPassword: z.string(),
```